content/master/concepts/composition-functions.md

```diff
@@ -1,1026 +1,699 @@
+line="11">}}input{{</hover>}}. Different Functions each have a different
+{{<hover label="single" line="13">}}kind{{</hover>}} of input. This example uses
+[Function Patch and Transform](https://github.com/crossplane-contrib/function-patch-and-transform).
+Function Patch and Transform is a function that implements Crossplane resource
+templates. Its input kind is `Resources`, and it accepts
+[Patch and Transform]({{<ref "./patch-and-transform">}}) {{<hover label="single"
+line="14">}}resources{{</hover>}} as input.
+
+```yaml {label="single",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  # Removed for Brevity
+  mode: Pipeline
+  pipeline:
+  - step: patch-and-transform
+    functionRef:
+      name: function-patch-and-transform
+    input:
+      apiVersion: pt.fn.crossplane.io/v1beta1
+      kind: Resources
+      resources:
+      - name: storage-bucket
+        base:
+          apiVersion: s3.aws.upbound.io/v1beta1
+          kind: Bucket
+          spec:
+            forProvider:
+              region: "us-east-2"
+```
 
-Your XRs, claims, and providers don't need to be updated or otherwise aware
-of Composition Functions to use them. They need only use a `Composition` that
-includes one or more entries in its `spec.functions` array.
+{{<hint "important">}}
+You can't specify resource templates using the Composition's `spec.resources`
+field when you use composition functions. When a Composition's `spec.mode` is
+`Pipeline` you can only specify `spec.pipeline`.
+{{< /hint >}}
 
-Composition Functions are designed to be run in a pipeline, so you can 'stack'
-several of them together. Each Function is passed the output of the previous
-Function as its input. Functions can also be used in conjunction with P&T
-Composition (a `spec.resources` array). 
+## Use a pipeline of functions in a composition
 
-In the following example P&T Composition composes an RDS instance. A pipeline of
-(hypothetical) Composition Functions then mutates the desired RDS instance by
-adding a randomly generated password, and composes an RDS security group.
+Crossplane can ask more than one Function what to do when a composite resource
+changes. When a Composition has a pipeline of two or more steps, Crossplane
+calls them all. It calls them in the order they appear in the pipeline.
 
-```yaml
+<!-- vale Microsoft.Auto = NO -->
+<!--
+  This linter does not allow "auto-ready", which is part of the name of the
+  function.
+-->
+Crossplane passes each Function in the pipeline the result of the previous
+Function. This enables powerful combinations of Functions. In this example,
+Crossplane calls {{<hover label="double" line="10">}}function-cue{{</hover>}} to
+create an S3 bucket. Crossplane then passes the bucket to {{<hover
+label="double" line="23">}}function-auto-ready{{</hover>}}, which marks the
+composite resource as ready when the bucket becomes ready.
+<!-- vale Microsoft.Auto = YES -->
```

Comment:
```suggestion
Crossplane passes each Function in the pipeline the result of the previous
Function. This enables powerful combinations of Functions. In this example,
Crossplane calls {{<hover label="double" line="10">}}function-cue{{</hover>}} to
create an S3 bucket. Crossplane then passes the bucket to 
{{<hover label="double" line="23">}}function-auto-ready{{</hover>}}, which marks the
composite resource as ready when the bucket becomes ready.
```

Vale ignores the contents of `{{<hover >}}` elements but it has to be on a single line